Source/Prism.Ninject.Wpf.Tests/NinjectRegionNavigationContentLoaderFixture.cs

```diff
@@ -0,0 +1,72 @@
+using System.Linq;
+using Microsoft.Practices.ServiceLocation;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Ninject;
+using Prism.IocContainer.Wpf.Tests.Support.Mocks.Views;
+using Prism.Ninject;
+using Prism.Ninject.Wpf.Tests.Mocks;
+using Prism.Regions;
+
+namespace Prism.Unity.Wpf.Tests
+{
+    [TestClass]
+    public class NinjectRegionNavigationContentLoaderFixture
+    {
+        private MockServiceLocator mockServiceLocator;
+        private IKernel _kernel;
+
+        [TestMethod]
+        public void ShouldFindCandidateViewInRegion()
+        {
+            IKernel kernel = new StandardKernel();
+            kernel.Bind<object, MockView>().To<MockView>().Named("MockView");
+
+            ConfigureMockServiceLocator(kernel);
+
+            // We cannot access the UnityRegionNavigationContentLoader directly so we need to call its
+            // GetCandidatesFromRegion method through a navigation request.
+            IRegion testRegion = new Region();
+
+            var view = new MockView();
+            testRegion.Add(view);
+            testRegion.Deactivate(view);
+
+            testRegion.RequestNavigate("MockView");
+
+            Assert.IsTrue(testRegion.Views.Contains(view));
+            Assert.IsTrue(testRegion.Views.Count() == 1);
+            Assert.IsTrue(testRegion.ActiveViews.Count() == 1);
+            Assert.IsTrue(testRegion.ActiveViews.Contains(view));
+        }
+
+        [TestMethod]
+        public void ShouldFindCandidateViewWithFriendlyNameInRegion()
+        {
+            IKernel kernel = new StandardKernel();
+            kernel.RegisterTypeForNavigation<MockView>("SomeView");
+         //   kernel.Bind<object, MockView>().To<MockView>().Named("SomeView");
```

Comment:
if this is an explanation of what happens in ninject, place it behind the previous line. Otherwise it can be deleted to prevent confusion.

Reply:
I was toying around with something else to see if I could get the type checking working through RegisterTypeForNavigation.  I'll use the commented out line for now rather than the RegisterTypeForNavigation as that's what the other tests appear to use.